domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,142 @@
+package org.oppia.android.domain.platformparameter
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+const val stringPlatformParameterName = "string_platform_parameter_name"
+const val stringPlatformParameterValue = "string_platform_parameter_value"
+
+const val integerPlatformParameterName = "integer_platform_parameter_name"
+const val integerPlatformParameterValue = 1
+
+const val booleanPlatformParameterName = "boolean_platform_parameter_name"
+const val booleanPlatformParameterValue = true
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveStringParameter_verifyItsValue() {
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
```

Comment:
This can also be it's own test `testSingleton_init/setPlatformParameterMap_isNotEmpty()` and then testing for each value type can just check the value.

Reply:
Sure will do that